docs/ai/how-to/app-service-entra-auth.md

```diff
@@ -0,0 +1,155 @@
+author: haywoodsloan
+ms.topic: how-to
+ms.date: 04/19/2024
+zone_pivot_groups: azure-development-tool-set-two
+
+#customer intent: As a .NET developer, I want authenticate and authorize my App Service to Azure OpenAI using Microsoft Entra so that I can securely use AI in my .NET application.
+
+---
+
+# Authenticate and Authorize App Service to Azure OpenAI using Microsoft Entra and the Semantic Kernel SDK
+
+This article demonstrates how to use Microsoft Entra managed identities to authenticate and authorize an App Service application to an Azure OpenAI resource.
+
+Using a managed identity from Microsoft Entra allows your App Service application to easily access a protected Azure OpenAI resources without having to provision or rotate any secrets.
+
+The [Semantic Kernel SDK](/semantic-kernel/overview) enables you to easily implement Microsoft Entra authentication in your .NET application.
+
+## Prerequisites
+
+* An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+* [.NET SDK](https://dotnet.microsoft.com/download/visual-studio-sdks)
+* [`Microsoft.SemanticKernel` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel)
+* [`Azure.Identity` NuGet package](https://www.nuget.org/packages/Azure.Identity)
+* [Create and deploy an Azure OpenAI Service resource](/azure/ai-services/openai/how-to/create-resource)
+* [Create and deploy a .NET application to App Service](/azure/app-service/quickstart-dotnetcore)
+
+## Add a managed identity to App Service
+
+### Add a system-assigned identity
+
+:::zone target="docs" pivot="azure-portal"
+
+1. Navigate to your app's page on the [Azure Portal](https://aka.ms/azureportal), scroll down to the **Settings** group.
+1. Select **Identity**.
+1. Within the **System assigned** tab, switch **Status** to On. Click **Save**.
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+Run the az webapp identity assign command to create a system-assigned identity:
+
+```azurecli
+az webapp identity assign --name <app-name> --resource-group <group-name>
+```
+
+:::zone-end
+
+### Add a user-assigned identity
+
+Creating an app with a user-assigned identity requires that you create the identity and then add its resource identifier to your app config.
+
+:::zone target="docs" pivot="azure-portal"
+
+1. Create a user-assigned managed identity resource according to these [instructions](/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal#create-a-user-assigned-managed-identity).
+1. In the left navigation for your app's page, scroll down to the **Settings** group.
+1. Select **Identity**.
+1. Select **User assigned > Add**.
+1. Search for the identity you created earlier, select it, and select **Add**.
+
+    > [!NOTE]
```

Comment:
It's listed outside the original instruction set I pulled this info from: https://learn.microsoft.com/en-us/azure/app-service/overview-managed-identity?tabs=portal%2Chttp#add-a-system-assigned-identity.

Since the goal of these steps is to add the user-assigned identity, the restart is just a side-effect. So I included this as a note since it's important info to be aware of but not required for the intended goal of these instructions.

I could see making this a warning instead, because my intention with putting this into a callout is to make sure it draws attention.